runtime/src/in_mem_accounts_index.rs

```diff
@@ -946,55 +943,79 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
         self.stats()
             .estimate_mem
             .store(estimate_mem as u64, Ordering::Relaxed);
+        exceeds_budget
+    }
+
+    /// scan loop
+    /// holds read lock
+    fn flush_scan(&self, current_age: Age, startup: bool) -> FlushScanResult<T> {
```

Comment:
How about also passing in the `&FlushGuard` to this fn too? Since it's a borrow, I think it should be OK.